### PR TITLE
Define difference in build/publish

### PR DESCRIPTION
**Pull request by drGrove**
PRs should check that the image builds while pushes to master should
acutally generate an image. This also is needed because dockerhub has
instilled rate limits meaning that we must have a bot user with
credentials. To prevent leaks this bot user has no actual access to
hashbang builds and can only pull public images.

### CODE REVIEW

**Review by daurnimator — COMMENTED** [review]
the two steps seem to do the same thing? oh; the `dry_run` flag.
But did you intend for them to have different credentials?

File: .drone.yml [hunk 1]
```diff
@@ -1,10 +1,48 @@
...
+      username: hashbangbot
+      password:
+        from_secret: dockerhub_ro_passphrase
+      purge: true
+      dry_run: true
+
+trigger:
+  event:
+    - pull_request
+---
```
Comment (daurnimator) [inline .drone.yml:28]: maybe add blank line between sections?

**Review by daurnimator — COMMENTED** [review]
perhaps time to break out the jsonnet?

File: .drone.yml [hunk 2]
```diff
@@ -12,7 +13,41 @@ steps:
       recursive: true
       tags: true
   - name: build
-    image: plugins/docker
+    image: banzaicloud/drone-kaniko
```
Comment (daurnimator) [inline .drone.yml]: Fails with
> error checking push permissions -- make sure you entered the correct tag name, and that you are authenticated correctly, and try again: checking push permission for "index.docker.io/hashbang/book:latest": UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:hashbang/book Type:repository] map[Action:push Class: Name:hashbang/book Type:repository]]

seems like this image ignores `dry_run`

**Comment by drGrove** [comment]
> the two steps seem to do the same thing? oh; the `dry_run` flag.
> But did you intend for them to have different credentials?

Yes. I don't want something that has write access to our registry running in PRs. I've created a second bot user that only has access to public repos

**Comment by drGrove** [comment]
> perhaps time to break out the jsonnet?

Probably, mostly just trying to get builds working first

File: .drone.yml [hunk 3]
```diff
@@ -12,11 +13,48 @@ steps:
       recursive: true
       tags: true
   - name: build
-    image: plugins/docker
+    image: drgrove/drone-kaniko@sha256:e3045421c3683e6baf5628b22ea0ee1cd7ae217f4de0e1bc53a0a1a20335b108
```
Comment (daurnimator) [inline .drone.yml:17]: why your fork?
Please at least link the commit this image is from in a comment

**Review by daurnimator — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 3 above)
Comment (drGrove) [reply .drone.yml:17]: They archived the repo and haven't updated it in 16 months

File: .drone.yml (hunk 1 above)
Comment (drGrove) [reply .drone.yml:28]: When I run the sign it doesn't put a blank line before that second. So I'll assume that most tooling would probably strip the newline

**Review by daurnimator — APPROVED** [review]

**Pull request merged**